test/integration/ocsp_test.go

```diff
@@ -0,0 +1,192 @@
+// +build integration
+
+package integration
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/eggsampler/acme/v2"
+	ocsp_helper "github.com/letsencrypt/boulder/test/ocsp/helper"
+)
+
+func random_domain() string {
+	var bytes [3]byte
+	rand.Read(bytes[:])
+	return hex.EncodeToString(bytes[:]) + ".com"
+}
+
+type client struct {
+	acme.Account
+	acme.Client
+}
+
+func makeClient() (*client, error) {
+	c, err := acme.NewClient(os.Getenv("DIRECTORY"))
+	if err != nil {
+		return nil, fmt.Errorf("Error connecting to acme directory: %v", err)
+	}
+	privKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		return nil, fmt.Errorf("error creating private key: %v", err)
+	}
+	account, err := c.NewAccount(privKey, false, true, "mailto:example@letsencrypt.org")
+	if err != nil {
+		return nil, fmt.Errorf("error creating new account: %v", err)
+	}
+	return &client{account, c}, nil
+}
+
+func addHTTP01Response(token, keyAuthorization string) error {
+	resp, err := http.Post("http://boulder:8055/add-http01", "",
+		bytes.NewBufferString(fmt.Sprintf(`{
+		"token": "%s",
+		"content": "%s"
+	}`, token, keyAuthorization)))
+	if err != nil {
+		return fmt.Errorf("adding http-01 response: %s", err)
+	}
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("adding http-01 response: status %d", resp.StatusCode)
+	}
+	resp.Body.Close()
+	return nil
+}
+
+type issuanceResult struct {
+	acme.Order
+	certs []*x509.Certificate
+}
+
+func authAndIssue(domains []string) (*issuanceResult, error) {
+	c, err := makeClient()
+	if err != nil {
+		return nil, err
+	}
+	var ids []acme.Identifier
+	for _, domain := range domains {
+		ids = append(ids, acme.Identifier{Type: "dns", Value: domain})
+	}
+	order, err := c.Client.NewOrder(c.Account, ids)
+	if err != nil {
+		return nil, fmt.Errorf("making order: %s", err)
+	}
+
+	for _, authUrl := range order.Authorizations {
+		auth, err := c.Client.FetchAuthorization(c.Account, authUrl)
+		if err != nil {
+			return nil, fmt.Errorf("fetching authorization at %s: %s", authUrl, err)
+		}
+
+		chal, ok := auth.ChallengeMap[acme.ChallengeTypeHTTP01]
+		if !ok {
+			return nil, fmt.Errorf("no HTTP challenge at %s", authUrl)
+		}
+
+		err = addHTTP01Response(chal.Token, chal.KeyAuthorization)
+		if err != nil {
+			return nil, fmt.Errorf("adding HTTP-01 response: %s", err)
+		}
+		chal, err = c.Client.UpdateChallenge(c.Account, chal)
+		if err != nil {
+			return nil, fmt.Errorf("updating challenge: %s", err)
+		}
+	}
+
+	certKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
```

Comment:
I think creating a CSR for a set of names with a private key (or a randomly generated one) will be a really common integration task test. Can you pull this out into a function? It will make the test body easier to read.